weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -904,6 +921,52 @@ def file_content_read(self, req: tsi.FileContentReadReq) -> tsi.FileContentReadR
             raise ValueError("Missing chunks")
         return tsi.FileContentReadRes(content=b"".join(chunks))
 
+    def feedback_create(
+        self, req: tsi.FeedbackCreateReqForInsert
+    ) -> tsi.FeedbackCreateRes:
+        feedback_id = generate_id()
+        created_at = datetime.datetime.now(ZoneInfo("UTC"))
+        # TODO: Any validation on weave_ref?
+        # TODO: What should max payload size be?
+        payload = _dict_value_to_dump(req.payload)
+        if len(payload) > 1024:
+            raise ValueError("Feedback payload too large")
+        row: Row = {
+            "id": feedback_id,
+            "project_id": req.project_id,
+            "weave_ref": req.weave_ref,
+            "wb_user_id": req.wb_user_id,
+            "creator": req.creator,
+            "feedback_type": req.feedback_type,
+            "payload": payload,
+            "created_at": created_at,
+        }
+        TABLE_FEEDBACK.insert(row).execute(self.ch_client)
+        return tsi.FeedbackCreateRes(
+            id=feedback_id, created_at=created_at, wb_user_id=req.wb_user_id
+        )
+
+    def feedback_query(self, req: tsi.FeedbackQueryReq) -> tsi.FeedbackQueryRes:
+        query = TABLE_FEEDBACK.select()
+        query = query.project_id(req.project_id)
+        query = query.fields(req.fields)
+        query = query.where(req.query)
+        query = query.order_by(req.sort_by)
+        query = query.limit(req.limit).offset(req.offset)
+        result = query.execute(self.ch_client)
+        return tsi.FeedbackQueryRes(result=result)
+
+    def feedback_purge(self, req: tsi.FeedbackPurgeReq) -> tsi.FeedbackPurgeRes:
```

Comment:
This feels extremely risky (a user would be able to purge an entire project's feedback in 1 call?

Reply:
That was a concern for me too. Batch deletion seems useful, but we could make it so we only allow purging a list of ids. We could also keep this as is but make the Python API more restrictive.

Reply:
i think we might want to heavily restrict this to a list of ids for now